iModelCore/ECDb/ECDb/SchemaWriter.cpp

```diff
@@ -286,6 +286,15 @@ BentleyStatus SchemaWriter::ImportSchema(Context& ctx, ECN::ECSchemaCR ecSchema)
                 {
                 if (schema->GetName().Equals(schemaChange->GetChangeName()))
                     {
+                    if (schemaChange->VersionMinor().GetOld() == schemaChange->VersionMinor().GetNew() && 

```

Comment:
We should be able to use SchemaKey::CompareByVersion or ::Matches here instead of putting our own check.

Reply:
@ColinKerr please have a quick look. I am surprised the SchemaComparer detects this as "modified" I thought we had an optimization so we would not compare schemas if version numbers are identical, so we don't do unnecessary work.... But seems like we do compare them.

Are you okay changing the behavior in general (so ECDb refuses to import schemas with matching version number but different contents)?

Reply:
> We should be able to use SchemaKey::CompareByVersion or ::Matches here instead of putting our own check.

This is a `SchemaChange` class which doesn't expose SchemaKeys. However I tackled it how it's usually done given the needed circumstances.

Reply:
> @ColinKerr please have a quick look. I am surprised the SchemaComparer detects this as "modified" I thought we had an optimization so we would not compare schemas if version numbers are identical, so we don't do unnecessary work.... But seems like we do compare them.
> 
> Are you okay changing the behavior in general (so ECDb refuses to import schemas with matching version number but different contents)?

We had discussed it over a call last week. It was decided to note the user of this error by logging an error, but still continue processing so that it doesn't disrupt any current workflows.

I also have noticed that if some major version bump requiring change is present, the schema import alreqdy fails, so some guards are already present - it would just alert about all of the expected minor increments as wel.